vllm/lora/fully_sharded_layers.py

```diff
@@ -0,0 +1,119 @@
+# pylint: disable=unused-argument
+from typing import TYPE_CHECKING, Optional
+
+import torch
+
+from vllm.lora.punica import bgmv, dispatch_bgmv_low_level
+from vllm.model_executor.parallel_utils.communication_op import (
+    tensor_model_parallel_all_gather,
+    tensor_model_parallel_all_reduce,
+)
+
+from vllm.lora.layers import (ColumnParallelLinearWithLoRA,
+                              MergedColumnParallelLinearWithLoRA,
+                              QKVParallelLinearWithLora,
+                              RowParallelLinearWithLoRA)
+
+if TYPE_CHECKING:
+    pass
+
+
+class ColumnParallelLinearWithShardedLoRA(ColumnParallelLinearWithLoRA):
```

Comment:
can we add docstrings to all classes? it's fine if they are 90% the same thing as the base class, but we should point out the differences